tmt/steps/provision/connect.py

```diff
@@ -127,6 +130,14 @@ def go(self) -> None:
             else:
                 data.key = [key]
 
+        if ssh_option:
+            for option in ssh_option:
+                self.info('ssh option', option, 'green')
+            self.debug('ssh option provided.')
+            data['ssh_option'] = ssh_option
```

Comment:
1. `data` is not a dictionary, but an instance of `ConnectGuestData` which is a subclass of `GuestSshData`.
2. `GuestSshData` lacks `ssh_option` field.

Reply:
3. two other plugins do ignore `--ssh-options`, is that correct?

Reply:
3. which two? 
Connection uses ssh, artemis uses ssh, minute tip uses ssh, podman uses something different.
I'm not sure about local plugin.

Reply:
1. & 2. not sure how to solve it now. Looks like there were some changes in codebase during development of this change.

Reply:
WRT 1. `data.ssh_option = ssh_option` should be the flavor of the week.

WRT 2. you need to add `ssh_option` field to `provision.GuestSshData` class, probably as `ssh_option: List[str] = dataclasses.field(default_class=list)`. All SSH-backed guests should then inherit it, with a nice empty list as a default value. We could even do `data.ssh_option = self.get('ssh-option')`, you can then check and/or iterate over `data.ssh_option`, no need for extra variable. And the same one to `GuestSsh` class itself, but with not default value.

WRT 3., `testcloud` and `artemis` - both do now have the `--ssh-option` option, and `testcloud` uses it thanks to        
 magic in `data = TestcloudGuestData(**{key: self.get(key) for key in TestcloudGuestData.keys()})`, but `artemis` plugin ignores it. I suppose it needs a similar bit to the one you added to the `connect` plugin right here.

WRT logging, is it needed? Guest data will be logged later anyway, and we would need to repeate this particular bit in all SSH-backed plugins to stay consistent (right now, `connect` uses `--ssh-option` and logs its content, `testcloud` uses it but logs nothing, `artemis` ignores it completely). So we need something as simple as possible, I'd start with `data.ssh_option = self.get('ssh-option')` with no logging involved, and we would then  - next patch? - add some shared solution, transparent for all plugins, like something happening in `Guest.__init__` or similar.